src/features/calendar/components/index.tsx

```diff
@@ -19,43 +24,82 @@ const Calendar = () => {
     TimeScale.MONTH
   );
 
+  const { orgId } = useRouter().query;
+  const model = useModel(
+    (env) => new CampaignActivitiesModel(env, parseInt(orgId as string))
+  );
+
   return (
-    <Box display="flex" flexDirection="column" height={'100%'} padding={2}>
-      <CalendarNavBar
-        focusDate={focusDate}
-        onChangeFocusDate={(date) => {
-          setFocusDate(date);
-        }}
-        onChangeTimeScale={(timeScale) => {
-          setSelectedTimeScale(timeScale);
-        }}
-        onStepBackward={() => {
-          setFocusDate(
-            dayjs(focusDate).subtract(1, selectedTimeScale).toDate()
-          );
-        }}
-        onStepForward={() => {
-          setFocusDate(dayjs(focusDate).add(1, selectedTimeScale).toDate());
-        }}
-        timeScale={selectedTimeScale}
-      />
-
-      <Box
-        display="flex"
-        flexDirection="column"
-        flexGrow={1}
-        marginTop={2}
-        overflow="auto"
-      >
-        {selectedTimeScale === TimeScale.DAY && <CalendarDayView />}
-        {selectedTimeScale === TimeScale.WEEK && (
-          <CalendarWeekView focusDate={focusDate} />
-        )}
-        {selectedTimeScale === TimeScale.MONTH && (
-          <CalendarMonthView focusDate={focusDate} />
-        )}
-      </Box>
-    </Box>
+    <ZUIFuture future={model.getAllActivities()}>
+      {(data) => {
+        const activitiesByDay = getActivitiesByDay(data);
+        const futureDays = getFutureDays(activitiesByDay, focusDate);
+        const lastDayWithEvent = getPreviousDay(activitiesByDay, focusDate);
```

Comment:
Nice! This will move to the server (RPC) eventually and I'm thinking we'll be able to reuse the code quite easily.